app/main/views/marketplace.py

```diff
@@ -42,11 +42,15 @@ def index():
             "framework {} status {}".format(framework.get('slug'), framework.get('status')))
         abort(500)
 
+    if current_user.is_authenticated():
+        logged_in_user = current_user.role
```

Comment:
I don't think you need to pass anything into the template at all here, as templates have access to `current_user`.

So, e.g. in _proposition_header.html we have `{% if current_user.email_address %}` and for the new links we can do in the template: `{% if current_user.role == 'supplier' %}` with no need for an additional parameter.